Fixes to build_release.sh script

mrflip opened the pull request:
- maven is putting the `storm-*` jar files in target/release+provided, not target/provided. Fixed path.
- storm-netty had a dependency itself on storm-core, so the storm-core jar was in both _release and _release/lib. Removing the dupe jar from `lib/`
- storm-console-processor was letting clojure-1.2 meet its dependency list, making for two clojure jars in the tarball. Added a dependency to its `project.clj`.
- build_releases.sh should fail if the build_modules.sh script fails. Made build_modules.sh run with set -e, and made build_releases.sh fail if it does
- made pretty banners, made paths be properly quoted
- bin/ directory files copy with permissions intact. Also now remove build_release.sh and build_modules.sh, since they won't work
- rather than remove _release and *.zip, only removes the specific subdirectory and .zip file. If you run with `STORM_KEEP_RELEASE=true`, it will not delete the release dir at the end of script (only at beginning)

Commit:
Fixes to build_release.sh script
* storm-console-processor was letting clojure-1.2 meet its dependency list, making for two clojure jars in the tarball. Added a dependency to its project.clj
* storm-netty had a dependency itself on storm-core, so the storm-core jar was in both _release and _release/lib. removing the dupe jar from lib/
* build_releases should fail if the build_modules script fails. Made build_modules run set -e, and made build_releases fail if it does
* maven is putting the storm-* jar files in target/release+provided, not target/provided. Fixed path.
* made pretty banners, made paths be properly quoted
* bin/ directory files copy with permissions intact. Also now remove build_release.sh and build_modules.sh, since they won't work
* rather than remove _release and *.zip, only removes the specific subdirectory and .zip file. If you run with STORM_KEEP_RELEASE=true, it will not delete the release dir at the end of script (only at beginning)
* bin/storm tells you how to roll your own release if you try to launch from a git checkout

mrflip commented:
Just saw this overlaps with #597 -- let me know if I should rebase. (This patch implements the same functionality)

nathanmarz commented on bin/build_release.sh:
```diff
@@ -1,55 +1,80 @@
-	cd ..
+  cd $module
+  mvn dependency:copy-dependencies || exit 1
+  cp -f target/release/dependency/*.jar "$DIR/lib/"
```
For me these dirs don't exist, I just have target/dependency.

nathanmarz commented:
I tried running this and it doesn't work on my computer. No jars are copied to the release distribution.

mrflip commented:
Gah sounds like I'm gonna have to read some maven documentation. I'll update the pull request once I know why we get different destinations.

On Jul 9, 2013, at 1:03 PM, Nathan Marz notifications@github.com wrote:

> I tried running this and it doesn't work on my computer. No jars are copied to the release distribution.
> 
> —
> Reply to this email directly or view it on GitHub.

Commit:
Don't let leiningen override the target path with target/release
Recent leiningen (technomancy/leiningen@a1d7b6f3eabe4eb) appends the profile name to the target directory -- if you installed lein before may, you got a pom with build dir = target/; recent leins set it to target/{profile_name}. This expressly fixes it to be just plain 'target'.

mrflip commented:
Looks like technomancy/leiningen@a1d7b6f3eabe4eb made it append the profile name to the target directory -- if you installed lein before May, you got a pom with `<directory>target</directory>`; with recent lein, `<directory>target/release</directory>`. 

I made the project.clj file fix the directory as just plain `target`, and rebased on top of current master.
